spring-cloud-starter-stream-common-gemfire/src/main/java/org/springframework/cloud/stream/app/gemfire/config/GemfireSslProperties.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2018 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.gemfire.config;
+
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotBlank;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.core.io.Resource;
+import org.springframework.util.StringUtils;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * @author Christian Tzolov
+ */
+@ConfigurationProperties("gemfire.security.ssl")
+@Validated
+public class GemfireSslProperties {
+
+	private static final String USER_HOME_DIRECTORY = System.getProperty("user.home");
+	/**
+	 * Name of the local trust store file copied in the local file system.
+	 */
+	public static final String LOCAL_TRUSTSTORE_FILE_NAME = "trusted.keystore";
+	/**
+	 * Name of the local trust store file copied in the local file system.
+	 */
+	public static final String LOCAL_KEYSTORE_FILE_NAME = "keystore.keystore";
+
+	/**
+	 * Local directory to store the truststore and keystore files copied form the remote truststoreUri and getKeystoreUri uris.
+	 */
+	@NotBlank
+	private String userHomeDirectory = USER_HOME_DIRECTORY;
+
+	/**
+	 * Location of the pre-created truststore URI to be used for connecting to the Geode cluster
```

Comment:
Pay attention how all the configuration properties JavaDocs should end with the period and should not use JavaDoc tag, like you do in the `truststoreType` description.

That's Spring Boot requirements.